### PR TITLE
Fix: Increment beatmap playcount & passcount on score submission

### PR DESCRIPTION
**Pull request by cmyui**
Should also fix these:

image

### CODE REVIEW

File: app/usecases/beatmap.py [hunk 1]
```diff
@@ -298,12 +298,20 @@ def parse_from_osu_api(
...
     await app.state.services.database.execute(
-        "UPDATE beatmaps SET passcount = :pass, playcount = :play WHERE beatmap_md5 = :md5",
-        {"play": beatmap.plays, "pass": beatmap.passes, "md5": beatmap.md5},
+        """\
+        UPDATE beatmaps
+        SET passcount = passcount + 1,
+        playcount = playcount + 1
+        WHERE beatmap_md5 = :beatmap_md5
+        """,
+        {"beatmap_md5": beatmap.md5},
```
Comment (cmyui) [inline app/usecases/beatmap.py]: Decoupled from `beatmap` object to reduce risk in case it has incorrect data

File: app/api/score_sub.py [hunk 2]
```diff
@@ -354,6 +354,11 @@ async def submit_score(
 
     await app.usecases.stats.save(stats)
 
+    await app.usecases.beatmap.increment_playcount(
+        beatmap=beatmap,
+        increment_passcount=score.passed,
+    )
```
Comment (cmyui) [inline app/api/score_sub.py:360]: Seems like there are no cases `return`ing above where this would be desirable

Comment (cmyui) [reply app/api/score_sub.py:360]: Maybe this one is debatable:

https://github.com/osuAkatsuki/score-service/blob/9c0d449fbaeebc4b83745662d8b67badd70ae8cf/app/api/score_sub.py#L151-L152

**Review by infernalfire72 — APPROVED** [review]

**Review by infernalfire72 — DISMISSED** [review]

File: app/usecases/beatmap.py [hunk 3]
```diff
@@ -298,12 +298,21 @@ def parse_from_osu_api(
...
-    if passcount:
+    if increment_passcount:
         beatmap.passes += 1
 
     await app.state.services.database.execute(
-        "UPDATE beatmaps SET passcount = :pass, playcount = :play WHERE beatmap_md5 = :md5",
-        {"play": beatmap.plays, "pass": beatmap.passes, "md5": beatmap.md5},
+        """\
+        UPDATE beatmaps
+        SET passcount = passcount + 1,
```
Comment (infernalfire72) [inline app/usecases/beatmap.py]: this seems to be updating the passcount regardless of increment_passcount being set

Comment (cmyui) [reply app/usecases/beatmap.py]: ty

**Review by infernalfire72 — APPROVED** [review]

**Pull request merged**